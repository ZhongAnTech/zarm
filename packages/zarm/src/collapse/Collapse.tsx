import React, { HTMLAttributes, Component, Children, cloneElement, ReactElement } from 'react';
import classnames from 'classnames';
import { CollapseActiveKey, CollapseItemKey, BaseCollapseProps } from './PropsType';
import CollapseItem, { CollapseItemProps } from './CollapseItem';

export interface CollapseProps
  extends Omit<HTMLAttributes<HTMLDivElement>, 'activeKey' | 'defaultActiveKey' | 'onChange'>,
    BaseCollapseProps {
  prefixCls?: string;
}

interface CollapseStates {
  activeKey?: CollapseActiveKey;
  prevActiveKey?: CollapseActiveKey;
  animated?: boolean;
  multiple?: boolean;
}

const getActiveKey = (props: CollapseProps) => {
  const { multiple, activeKey, defaultActiveKey } = props;

  let value;

  if (typeof activeKey !== 'undefined') {
    value = activeKey;
  }

  if (!('activeKey' in props) && typeof defaultActiveKey !== 'undefined') {
    value = defaultActiveKey;
  }

  if (value) {
    return multiple ? [].concat(value) : value;
  }

  return multiple ? [] : undefined;
};

export default class Collapse extends Component<CollapseProps, CollapseStates> {
  static defaultProps: CollapseProps = {
    prefixCls: 'za-collapse',
    multiple: false,
    animated: false,
    onChange: () => {},
  };

  static Item: typeof CollapseItem;

  state: CollapseStates = {
    activeKey: getActiveKey(this.props),
  };

  static getDerivedStateFromProps(nextProps: CollapseProps, state: CollapseStates) {
    const newState: CollapseStates = {};

    if ('activeKey' in nextProps && nextProps.activeKey !== state.activeKey) {
      newState.activeKey = getActiveKey(nextProps);
    }

    if ('activeKey' in newState) {
      return newState;
    }

    return null;
  }

  onItemChange = (onItemChange, key) => {
    if (!key) {
      return;
    }

    let { activeKey } = this.state;
    const { onChange, multiple } = this.props;
    let isActive;
    let newActiveKey;

    if (multiple) {
      newActiveKey = [];
      activeKey = (activeKey as CollapseItemKey[]) || [];
      if (activeKey.indexOf(key) > -1) {
        newActiveKey = activeKey.filter((i) => i !== key);
      } else {
        newActiveKey = activeKey.slice(0);
        newActiveKey.push(key);
      }
      isActive = newActiveKey.indexOf(key) > -1;
    } else {
      activeKey = activeKey as CollapseItemKey;
      newActiveKey = activeKey === key ? undefined : key;
      isActive = activeKey === key;
    }

    if (!('activeKey' in this.props)) {
      this.setState({
        activeKey: newActiveKey,
      });
    }

    typeof onItemChange === 'function' && onItemChange(isActive);
    typeof onChange === 'function' && onChange(newActiveKey);
  };

  renderItems = () => {
    const { activeKey } = this.state;
    const { multiple, animated } = this.props;
    return Children.map(this.props.children, (ele: ReactElement<CollapseItemProps>) => {
      const { disabled, onChange } = ele.props;
      const { key } = ele;
      const isActive = multiple
        ? ((activeKey as CollapseItemKey[]) || []).indexOf(key!) > -1
        : (activeKey as CollapseItemKey) === key;

      return cloneElement(ele, {
        animated,
        isActive,
        onChange: () => !disabled && this.onItemChange(onChange, key),
      });
    });
  };

  render() {
    const {
      prefixCls,
      className,
      onChange,
      animated,
      activeKey,
      defaultActiveKey,
      ...rest
    } = this.props;

    const cls = classnames(prefixCls, className, {
      [`${prefixCls}--animated`]: animated,
    });

    return (
      <div className={cls} {...rest}>
        {this.renderItems()}
      </div>
    );
  }
}
