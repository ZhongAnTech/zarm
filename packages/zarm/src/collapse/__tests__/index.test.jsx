import React from 'react';
import { render, mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Collapse from '../index';

describe('Collapse', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <Collapse>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with collapse mode', () => {
    const props = {};
    props.multiple = true;
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with animated', () => {
    const props = {};
    props.animated = true;
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with defaultActiveKey', () => {
    const props = {};
    props.defaultActiveKey = '1';
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with number defaultActiveKey', () => {
    const props = {};
    props.defaultActiveKey = 0;
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with array defaultActiveKey', () => {
    const props = {};
    props.multiple = true;
    props.defaultActiveKey = ['1'];
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with activeKey', () => {
    const props = {};
    props.activeKey = '0';
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with number activeKey', () => {
    const props = {};
    props.activeKey = 1;
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item key={0} title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key={1} title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key={2} title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with dynamic activeKey', () => {
    const props = {};
    props.activeKey = '0';
    const wrapper = shallow(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    wrapper.setProps({ activeKey: '1' });
    expect(wrapper.state('activeKey')).toEqual('1');
  });

  it('renders correctly with dynamic array activeKey', () => {
    const props = {};
    props.multiple = true;
    props.activeKey = ['0'];
    const wrapper = shallow(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    wrapper.setProps({ activeKey: ['1'] });
    expect(wrapper.state('activeKey')).toEqual(['1']);
  });

  it('renders correctly with defaultActiveKey and activeKey', () => {
    const props = {};
    props.defaultActiveKey = '0';
    props.activeKey = '1';
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    expect(wrapper.find('.za-collapse-item--active').length).toBe(1);
  });

  it('click collapse item correctly', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Collapse onChange={onChange}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    wrapper.find('.za-collapse-item__title').at(0).simulate('click');
    expect(onChange).toBeCalled();
  });

  it('click collapse item correctly with disabled mode', () => {
    const props = {};
    props.onChange = jest.fn();
    const wrapper = mount(
      <Collapse {...props}>
        <Collapse.Item disabled key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    wrapper.find('.za-collapse-item__title').at(0).simulate('click');
    expect(props.onChange).not.toBeCalled();
  });

  it('collapse items toggle correctly with animated', (done) => {
    const props = {};
    props.animated = true;
    const wrapper = mount(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    wrapper.find('.za-collapse-item__title').at(0).simulate('click');
    setTimeout(() => {
      expect(wrapper.find('.za-collapse-item--active').length).toBe(1);
      done();
    }, 0);
  });

  it('negative item toggle correctly with animated', (done) => {
    const props = {};
    props.animated = true;
    const wrapper = mount(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    wrapper.find('.za-collapse-item__title').at(0).simulate('click');
    setTimeout(() => {
      expect(wrapper.find('.za-collapse-item--active').length).toBe(1);
      done();
    }, 0);
  });

  it('active item toggle correctly with animated', (done) => {
    const props = {};
    props.animated = true;
    props.defaultActiveKey = '0';
    const wrapper = mount(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    wrapper.find('.za-collapse-item__title').at(0).simulate('click');
    setTimeout(() => {
      expect(wrapper.find('.za-collapse-item--active').length).toBe(0);
      done();
    }, 0);
  });

  it('collapse items toggle correctly without multiple mode', () => {
    const props = {};
    props.multiple = false;
    props.activeKey = '1';
    props.onChange = jest.fn();
    const wrapper = mount(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    wrapper.find('.za-collapse-item__title').at(0).simulate('click');
    expect(wrapper.find('.za-collapse-item--active').length).toBe(1);
  });

  it('collapse items toggle correctly with multiple mode', () => {
    const props = {};
    props.multiple = true;
    props.defaultActiveKey = ['0', '1'];

    const wrapper = mount(
      <Collapse {...props}>
        <Collapse.Item key="0" title="Header of Item1">
          This is content of item1.
        </Collapse.Item>
        <Collapse.Item key="1" title="Header of Item2">
          This is content of item2.
        </Collapse.Item>
        <Collapse.Item key="2" title="Header of Item3">
          This is content of item3.
        </Collapse.Item>
      </Collapse>,
    );
    wrapper.find('.za-collapse-item__title').at(0).simulate('click');
    expect(wrapper.find('.za-collapse-item--active').length).toBe(1);
  });

  it('click should not trigger callback without key', () => {
    const props = {};
    props.onChange = jest.fn();
    const wrapper = mount(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
      </Collapse>,
    );
    wrapper.find('.za-collapse-item__title').simulate('click');
    expect(props.onChange).not.toBeCalled();
  });
});
