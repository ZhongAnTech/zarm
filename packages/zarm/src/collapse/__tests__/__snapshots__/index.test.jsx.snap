// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse renders correctly 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item1
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item1.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item2
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item2.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item3
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item3.
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with activeKey 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item za-collapse-item--active"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item1
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item1.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item2
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item2.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item3
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item3.
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with animated 1`] = `
<div
  class="za-collapse za-collapse--animated"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item1
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item1.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item2
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item2.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item3
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item3.
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with array defaultActiveKey 1`] = `
<div
  class="za-collapse"
  multiple=""
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item1
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item1.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item za-collapse-item--active"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item2
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item2.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item3
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item3.
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with collapse mode 1`] = `
<div
  class="za-collapse"
  multiple=""
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item1
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item1.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item2
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item2.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item3
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item3.
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with defaultActiveKey 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item1
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item1.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item za-collapse-item--active"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item2
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item2.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item3
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item3.
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with number activeKey 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item1
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item1.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item2
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item2.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item3
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item3.
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with number defaultActiveKey 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item1
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item1.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item2
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item2.
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item__header"
    >
      <div
        class="za-collapse-item__title"
      >
        Header of Item3
      </div>
      <div
        class="za-collapse-item__arrow"
      />
    </div>
    <div
      class="za-collapse-item__content"
    >
      <div
        class="za-collapse-item__content__inner"
      >
        This is content of item3.
      </div>
    </div>
  </div>
</div>
`;
